docs/console-access.md

```diff
@@ -0,0 +1,340 @@
+
+The serial console service exposed by QEMU has no explicit authentication
+options. It relies entirely on the network transport level to secure access.
+
+When a proxy is used in between the tenant user and QEMU service, it is possible
+to provide further authentication mechanisms at the network transport level, that
+are protocol agnostic. The general concept is that each network service exposed
+by a VM has an associated shared secret.
+
+In a HTTP websockets proxy, the tenant user provides the shared secret in a
+cookie when connecting to the proxy. The proxy uses the cookie to identify which
+virtual machine to connect to, by doing a lookup against a table mapping secrets
+to hostname + port number pairs. Thus the secret serves as both an identification
+and authentication mechanism. It is still possible to pass through the protocol
+specific authentication mechanism to the tenant user, but more likely is that
+the protocol specific authentication mechanism would only be used between the
+proxy and the QEMU service, if at all. It would be sufficient to rely on client
+certificates for authentication between the proxy and QEMU.
+
+In a HTTP CONNECT proxy, the tenant can provide the shared secret in a cookie as
+with a websockets proxy, but it will only be used for authentication. The VM is
+still identified by the address associated with the CONNECT verb. Alternatively
+the shared secret can be used as a pseudo hostname with the CONNECT verb. In this
+case the secret is used to lookup the real hostname + port of the QEMU service,
+and thus acts as both an identification and authentication mechanism. The latter
+has negative implications for x509 certificate validation though, because the
+pseudo-hostname used for CONNECT must match the address encoded in the cert. Thus
+a new cert would be required each time the shared secret changes, which is
+impractical.
+
+When using shared secrets with a HTTP CONNECT proxy, the connection between the
+tenant user and the proxy must also be protected with TLS, otherwise the secret
+can be intercepted. This would in turn mean that the console payload is double-
+encrypted, once with TLS for the proxy connection, and again for the content
+tunnelled over the proxy. This is bad for performance of both the proxy nodes
+and tenant user clients. This makes it undesirable to use the shared secret auth
+with HTTP CONNECT proxies. It is better to delegate all authentication to the
+protocol level, which allows the connection to proxy to remain in clear text,
+and rely on encryption of the tunnelled traffic instead. Until QEMU gains support
+for fine grained access control based on client certs, this only works for the
+SPICE and VNC services, since serial consoles have no native support for
+direct authentication.
+
+The shared secret used for proxy authentication can be treated as a long term
+persistent token, a time limited token, or a use limited (one-time) token. If
```

Comment:
I guess we have to go with one-tokens. This means that we can use the kubernetes security capabilities to retrieve the token and then use it.